gora-aerospike/src/main/java/org/apache/gora/aerospike/store/AerospikeStore.java

```
@@ -159,17 +167,25 @@ public boolean schemaExists() {
    * @return the Object corresponding to the key or null if it cannot be found
    */
   @Override
-  public T get(K key, String[] fields) {
-
-    Key recordKey = getAerospikeKey(key);
-    fields = getFieldsToQuery(fields);
+  public T get(K key, String[] fields) throws GoraException {
 
-    Record record = aerospikeClient
-            .get(aerospikeParameters.getAerospikeMapping().getReadPolicy(), recordKey, fields);
-    if (record == null) {
-      return null;
+    try {
+      Key recordKey = getAerospikeKey(key);
+      fields = getFieldsToQuery(fields);
+  
+      Record record = aerospikeClient
+              .get(aerospikeParameters.getAerospikeMapping().getReadPolicy(), recordKey, fields);
+      
+      if (record == null) {
+        return null;
+      }
+      return createPersistentInstance(record, fields);
+    } catch (GoraException e) {
+      throw e;

```

Hi,
Is there any specific reason why we are capturing the GoraException separately?

---

Hi! Yes. Notice that createPersistentInstance has been updated to throw GoraException. Throughout the changes, GoraException is being used as a simple wrapper to have a common interface for all datastores, so as you can see two lines bellow in this diff, any exception (not GoraException) is logged and wrapped. In the case of a catch of a GoraException we can safely assume that it is already logged and wraping other exception (or maybe an actual GoraException but this does not make difference) and we wouldn't want to wrap it again because that would be useless and would hide the exception's parent exception in an indefinite sequence.
Maybe it is not the best solution, but I didn't get to any better. If you have any suggestion it will be welcome!

---

Hi, Nishadi.
I have been thinking about this. You asked because in the case the is only a log and a throw? I actually just enclosed with try-catch whatever existed before without thinking much about it. But if you are saying that we should only throw the exceptions without logging -since we are not logging anything interesting-, then I guess you are right. I could have done that, but I just did not want to change much the existing datastores :P My fault. It would be good just throw the exceptions and nothing more.
I will see about changing it whenever I have the time and energy :) Or if anyone wants to change it...

Thanks!